### PR TITLE
Deprecate SbThreadSleep

### PR DESCRIPTION
**Pull request by y4vor**
b/334139073

Change-Id: I8cb0562619e5e0a2392536d1637c3abd8ed326d8

### CODE REVIEW

**Comment by andrewsavage1** [comment]
Please fix the typo in the title of the PR

**Comment by y4vor** [comment]
> Please fix the typo in the title of the PR

done

File: starboard/shared/starboard/net_args.cc [hunk 1]
```diff
@@ -14,6 +14,7 @@
 
 #include "starboard/shared/starboard/net_args.h"
 
+#include <unistd.h>
```
Comment (andrewsavage1) [inline starboard/shared/starboard/net_args.cc:17]: newline

File: starboard/shared/starboard/audio_sink/stub_audio_sink_type.cc [hunk 2]
```diff
@@ -14,6 +14,7 @@
 
 #include "starboard/shared/starboard/audio_sink/stub_audio_sink_type.h"
 
+#include <unistd.h>
```
Comment (andrewsavage1) [inline starboard/shared/starboard/audio_sink/stub_audio_sink_type.cc:17]: newline

File: starboard/shared/starboard/player/filter/testing/adaptive_audio_decoder_test.cc [hunk 3]
```diff
@@ -12,6 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include <unistd.h>
```
Comment (andrewsavage1) [inline starboard/shared/starboard/player/filter/testing/adaptive_audio_decoder_test.cc:15]: newlie

File: starboard/shared/starboard/player/filter/testing/test_util.cc [hunk 4]
```diff
@@ -14,6 +14,7 @@
 
 #include "starboard/shared/starboard/player/filter/testing/test_util.h"
 
+#include <unistd.h>
```
Comment (andrewsavage1) [inline starboard/shared/starboard/player/filter/testing/test_util.cc:17]: separate with newline

File: starboard/shared/starboard/player/job_thread_test.cc [hunk 5]
```diff
@@ -14,6 +14,7 @@
 
 #include "starboard/shared/starboard/player/job_thread.h"
 
+#include <unistd.h>
```
Comment (andrewsavage1) [inline starboard/shared/starboard/player/job_thread_test.cc:17]: newline

File: starboard/shared/uwp/watchdog_log.cc [hunk 6]
```diff
@@ -14,6 +14,7 @@
 
 #include "starboard/shared/uwp/watchdog_log.h"
 
+#include <unistd.h>
```
Comment (andrewsavage1) [inline starboard/shared/uwp/watchdog_log.cc:17]: newline

File: third_party/boringssl/src/crypto/thread_starboard.cc [hunk 7]
```diff
@@ -16,8 +16,9 @@
 
 #include "internal.h"
 
-#if defined(STARBOARD)
+#include <unistd.h>
```
Comment (andrewsavage1) [inline third_party/boringssl/src/crypto/thread_starboard.cc:19]: The include should probably go in the `#if defined(STARBOARD)` block

File: third_party/llvm-project/libcxx/src/thread.cpp [hunk 8]
```diff
@@ -75,7 +75,8 @@ thread::detach()
 unsigned
 thread::hardware_concurrency() noexcept
 {
-#if defined(_SC_NPROCESSORS_ONLN)
+// Excluded from Modular builds as Starboard doesn't provide sysconf
+#if defined(_SC_NPROCESSORS_ONLN) && !SB_IS(MODULAR)
```
Comment (andrewsavage1) [inline third_party/llvm-project/libcxx/src/thread.cpp:79]: Why is this change necessary in this PR?

File: starboard/shared/alsa/alsa_audio_sink_type.cc [hunk 9]
```diff
@@ -16,6 +16,7 @@
 
 #include <alsa/asoundlib.h>
 
+#include <unistd.h>
```
Comment (andrewsavage1) [inline starboard/shared/alsa/alsa_audio_sink_type.cc:19]: newline

File: starboard/raspi/shared/open_max/video_decoder.cc [hunk 10]
```diff
@@ -12,6 +12,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include <unistd.h>
```
Comment (andrewsavage1) [inline starboard/raspi/shared/open_max/video_decoder.cc:15]: The c include should come after the class header

File: starboard/nplb/thread_test.cc [hunk 11]
```diff
@@ -12,6 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include <unistd.h>
```
Comment (andrewsavage1) [inline starboard/nplb/thread_test.cc:15]: newline

File: starboard/nplb/socket_send_to_test.cc [hunk 12]
```diff
@@ -15,6 +15,7 @@
 // SendTo is largely tested with ReceiveFrom, so look there for more involved
 // tests.
 
+#include <unistd.h>
```
Comment (andrewsavage1) [inline starboard/nplb/socket_send_to_test.cc:18]: newline

File: starboard/nplb/posix_compliance/posix_socket_send_test.cc [hunk 13]
```diff
@@ -16,6 +16,7 @@
 // this is hooked up to something.
 
 #include <fcntl.h>
+#include <unistd.h>
```
Comment (andrewsavage1) [inline starboard/nplb/posix_compliance/posix_socket_send_test.cc:19]: newline

File: starboard/nplb/audio_sink_test.cc [hunk 14]
```diff
@@ -14,6 +14,7 @@
 
 #include "starboard/audio_sink.h"
 
+#include <unistd.h>
```
Comment (andrewsavage1) [inline starboard/nplb/audio_sink_test.cc:17]: newline

File: starboard/android/shared/audio_track_audio_sink_type.cc [hunk 15]
```diff
@@ -14,6 +14,7 @@
 
 #include "starboard/android/shared/audio_track_audio_sink_type.h"
 
+#include <unistd.h>
```
Comment (andrewsavage1) [inline starboard/android/shared/audio_track_audio_sink_type.cc:17]: c headers and c++ headers should be separated by a newline

File: third_party/llvm-project/libcxx/src/thread.cpp (hunk 8 above)
Comment (y4vor) [reply third_party/llvm-project/libcxx/src/thread.cpp:79]: The _SC_NPROCESSORS_ONLN is defined in <unistd.h> which triggers sysconf.
In this PR we start using <unistd.h> for `usleep`.

File: starboard/nplb/posix_compliance/posix_socket_sendto_test.cc [hunk 16]
```diff
@@ -16,6 +16,7 @@
 // this is hooked up to something.
 
 #include <fcntl.h>
+#include <unistd.h>
```
Comment (andrewsavage1) [inline starboard/nplb/posix_compliance/posix_socket_sendto_test.cc:19]: newline

File: third_party/boringssl/src/crypto/thread_starboard.cc [hunk 17]
```diff
@@ -16,7 +16,9 @@
 
 #include "internal.h"
 
+
```
Comment (andrewsavage1) [inline third_party/boringssl/src/crypto/thread_starboard.cc:19]: remove extra newline

**Review by andrewsavage1 — APPROVED** [review]
